### ISSUE TITLE
`sudo nixos-rebuild edit` doesn't use default editor

### ISSUE THREAD

**Issue by awall** [NotAShelf/neovim-flake#80]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked all existing issues and didn't find a similar issue

### Description

When I run `sudo nixos-rebuild edit`, it always opens in nano, even though neovim is the default editor. Running `nix-os-rebuild edit`, without `sudo`, opens in neovim as it should, but without sudo I can't actually save the file.

In `configuration.nix`, I have `programs.neovim.defaultEditor = true;`. That is the _global setting_, it is not specific to my user. If I run `sudo echo $EDITOR`, I get back `nvim`, so I'm not sure why it's confused...

### 👟 Reproduction steps

1. Add `programs.neovim.enabled = true;` and `programs.neovim.defaultEditor = true;` to `configuration.nix`.
2. Run `sudo nixos-rebuild switch`.
3. Reboot the machine.
4. In a prompt, type `sudo nixos-rebuild edit`.
5. **BUG: the file is opened in nano, instead of neovim.**

### 👀 Expected behavior

`sudo nixos-rebuild edit` should open `configuration.nix` in neovim.

### 😓 Actual Behavior

`sudo nixos-rebuild edit` opens `configuration.nix` in nano, regardless of my configuration.

### 💻 Metadata

this path will be fetched (0.01 MiB download, 0.04 MiB unpacked):   /nix/store/0jmdsgfnd6aakxdr0sl5l7zzfs59hdrw-stdenv-linux copying path '/nix/store/0jmdsgfnd6aakxdr0sl5l7zzfs59hdrw-stdenv-linux' from 'https://cache.nixos.org'...  - system: `"x86_64-linux"`  - host os: `Linux 5.15.110, NixOS, 22.11 (Raccoon), 22.11.4203.d41eea2839d`  - multi-user?: `yes`  - sandbox: `yes`  - version: `nix-env (Nix) 2.11.1`  - channels(root): `"home-manager-22.11.tar.gz, nixos-22.11"`  - channels(adam): `"home-manager-22.11.tar.gz"`  - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### 📝 Relevant log output

```bash
N/A
```

**Comment by NotAShelf** [comment]
`programs.neovim.enabled` is not a valid option, I assume you are referring to `programs.neovim.enable`. However that is also not provided by this flake. If you are using the home-manager module, sudo cannot use your neovim-flake installation because it installs programs to your user's profile - not system-wide.

**Comment by awall** [comment]
Makes sense, sorry I'm a bit a new to NixOS. What is the correct way to get it to use neovim for `sudo nixos-rebuilt edit`? Setting `environment.variables.EDITOR` has no effect, nor does setting the variable manually (`sudo export EDITOR=nvim`). This variable seems to be ignored by the root user.

**Comment by NotAShelf** [comment]
Home-manager makes packages available for the user they are configured for,
for instance:
 `home-manager.users.notashelf.home.packages = [ pkgs.neovim ];`
will make it available only for my user. So even if you set EDITOR variable globally, sudo will not be able to pick up what it corresponds to, "nvim", because neovim is not actually installed for the root user.

I understand you don't use this flake, but nixos' own neovim module. In that case you want to either add neovim to your `environment.systemPackages` or enable neovim module in nixos config rather than home-manager.

**Comment by awall** [comment]
Hmm, I think I just placed this bug report in the wrong project; this is a bug with nixos, but not this particular flake.

> In that case you want to either add neovim to your environment.systemPackages

I did this, no effect. I also updated the `environment.variables.EDITOR`, and again the root user seems to ignore all of this stuff. I'm making changes to the top-level config in my `etc/nixos/configuration.nix`, home manager has no part in this.
